app/src/ui/lib/author-input.tsx

```diff
@@ -743,18 +745,51 @@ export class AuthorInput extends React.Component<IAuthorInputProps, {}> {
       this.updateAuthors(cm)
     })
 
+    const wrapperElem = cm.getWrapperElement()
+
     // Do the very least we can do to pretend that we're a
     // single line textbox. Users can still paste newlines
     // though and if the do we don't care.
-    cm.getWrapperElement().addEventListener('keypress', (e: KeyboardEvent) => {
+    wrapperElem.addEventListener('keypress', (e: KeyboardEvent) => {
       if (!e.defaultPrevented && e.key === 'Enter') {
         e.preventDefault()
       }
     })
 
+    wrapperElem.addEventListener('contextmenu', e => {
+      this.onContextMenu(cm, e)
+    })
+
     return cm
   }
 
+  private onContextMenu(cm: Editor, e: PointerEvent) {
+    e.preventDefault()
+
+    const menu: IMenuItem[] = [
+      { label: 'Undo', action: () => cm.getDoc().undo() },
+      { label: 'Redo', action: () => cm.getDoc().redo() },
+      { type: 'separator' },
+      { role: 'cut' },
+      { role: 'copy' },
+      { role: 'paste' },
+    ]
+
+    if (__WIN32__) {
+      menu.push({ type: 'separator' })
+    }
+
+    menu.push({
+      label: __DARWIN__ ? 'Select All' : 'Select all',
+      action: () => {
+        // trust me, it's a thing :cry:
+        ;(cm as any).execCommand('selectAll')
```

Comment:
It looks like CodeMirror supports this:

```ts
        execCommand(name: string): void;
```

[source](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/6a187f6d4476403da23a2a77c3315d3bb573cc1b/types/codemirror/index.d.ts#L344)

Reply:
Lemme try updating our codemirror types